democracy/views/utils.py

```diff
@@ -97,3 +99,47 @@ def filter_by_hearing_visible(queryset, request, hearing_lookup='hearing'):
         filters['%s__published' % hearing_lookup] = True
 
     return queryset.filter(**filters)
+
+
+class JSONPKRelatedField(PrimaryKeyRelatedField):
+    """
+    Support of showing and saving of expanded JSON nesting or just a resource ID.
+    Serializing is controlled by query string param 'expand', deserialization
+    by format of JSON given.
+    Default serializing is expand=false.
+    """
+
+    invalid_json_error = _('Incorrect JSON. Expected JSON, received %(data)s.')
+    missing_id_error = _('The primary key is missing. Expected {"id": id, ...}, received %(data)s.')
+
+    def __init__(self, *args, **kwargs):
+        self.related_serializer = kwargs.pop('serializer', None)
+        self.expanded = bool(kwargs.pop('expanded', False))
+        super(JSONPKRelatedField, self).__init__(*args, **kwargs)
+
+    def use_pk_only_optimization(self):
+        return not self.expanded
+
+    def to_representation(self, obj):
+        if self.expanded:
+            return self.related_serializer(obj, context=self.context).data
+        id = super(JSONPKRelatedField, self).to_representation(obj)
+        if id is None:
+            return None
+        return {
+            'id': id,
+        }
+
+    def to_internal_value(self, value):
+        if not isinstance(value, dict):
+            raise ValidationError(self.invalid_json_error % {'data': type(value).__name__})
+        if 'id' not in value:
+            raise ValidationError(self.missing_id_error % {'data': value})
+
+        id = value['id']
+        if not id:
+            if self.required:
+                raise ValidationError(_('This field is required.'))
```

Comment:
Would be good if this specifies exactly which field is required, ie. raise `ValidationError({'id': [_('This field is required.')]})` or maybe use the `missing_id_error`?